api/src/test/java/org/openmrs/notification/MessageServiceTest.java

```diff
@@ -7,110 +7,294 @@
 	}
-	
-	/**
-	 * @throws MessageException
-	 * @see MessageService#sendMessage(Message)
-	 */
+
+    /**
+     * Tests that the message sender successfully sends a basic message.
+     *
+     * @throws Exception if sending fails
+     */
+    @Test
+    public void sendMessage_shouldSendMessageSuccessfully() throws Exception {
+        // Arrange
+        Message message = new Message();
+        message.setRecipients("recipient@example.com");
+        message.setSender("sender@example.com");
+        message.setSubject("Test Subject");
+        message.setContent("Test Content");
+
+        // Act
+        messageService.sendMessage(message);
+
+        // Assert
+        verify(mockSender, times(1)).send(message);
+    }
+
+    /**
+     * Tests sending a message to a user and verifies that the mock sender is invoked.
+     *
+     * @throws Exception if sending fails
+     */
 	@Test
-	public void sendMessage_shouldSendMessage() throws MessageException {
-		Message tryToSend1 = ms.createMessage("recipient@example.com", "sender@example.com", "subject", "content");
-		try {
-			ms.sendMessage(tryToSend1);
-		}
-		catch (MessageException e) {
-			//So that this test doesn't fail just because the user isn't running an SMTP server.
-			if (!e.getMessage().contains(NO_SMTP_SERVER_ERROR)) {
-				e.printStackTrace();
-				fail();
-			}
-		}
-		
-		Message tryToSend2 = ms.createMessage("recipient@example.com,recipient2@example.com", "openmrs.emailer@gmail.com",
-		    "subject", "content", "moo", "text/plain", "moo.txt");
-		try {
-			ms.sendMessage(tryToSend2);
-		}
-		catch (MessageException e) {
-			//So that this test doesn't fail just because the user isn't running an SMTP server.
-			if (!e.getMessage().contains(NO_SMTP_SERVER_ERROR)) {
-				e.printStackTrace();
-				fail();
-			}
-		}
-	}
-	
+    public void sendMessage_shouldSendMessageToUser() throws Exception {
+       // Arrange
+      Message message = new Message();
+      message.setRecipients("recipient@example.com");
+      message.setSender("sender@example.com");
+      message.setSubject("Hello");
+      message.setContent("Test message");
+
+    // Act
+    messageService.sendMessage(message);
+
+    // Assert: verify that the mock sender was used
+    verify(mockSender, times(1)).send(message);
+}
+
+ /**
+     * Ensures that the message service can prepare a message from a given template.
+     *
+     * @throws Exception if preparation fails
+     */
+@Test
+public void prepareMessage_shouldPrepareMessageFromTemplate() throws Exception {
+    Template template = new Template();
+
+    when(messagePreparator.prepare(any(Template.class)))
+         .thenAnswer(invocation -> new Message());
+
+
+    template.setName("Test Template");
+
+    Message expectedMessage = new Message();
+    expectedMessage.setRecipients("user@example.com");
+
+    // ✅ Tell the mock what to return
+    when(messagePreparator.prepare(template)).thenReturn(expectedMessage);
```

Comment:
Why do we retain the mock above if we shifted to this? Also, why have we dropped the arrange, act, assert structure here?